test/ERC1155PresetPausableNonTransferrable.test.ts

```diff
@@ -0,0 +1,289 @@
+import { expect, setupTest } from '@/test/helpers';
+import { createBatchMintData } from '@/test/helpers/erc-1155-preset-pausable-non-transferable';
+
+describe('ERC1155PresetPausableNonTransferrable', () => {
+  describe('mintBatch', () => {
+    it('should implement _beforeTokenTransfer', async () => {
+      const { mockERC1155PresetPausableNonTransferrable, hre } =
+        await setupTest();
+      await expect(
+        mockERC1155PresetPausableNonTransferrable.mintBatch(
+          hre.namedAccounts.buyer,
+          [0],
+          [0],
+          createBatchMintData({ amount: 1, hre })
+        )
+      ).to.emit(
+        mockERC1155PresetPausableNonTransferrable,
+        'BeforeTokenTransfer'
```

Comment:
i'm having trouble understanding how this check is valuable given that the event is just emitted from the mock contract directly... I guess it's just indicating that `_beforeTokenTransfer` on the mock is being called, which should only happen if the implementation of `mintBatch` in the real parent contract calls `_beforeTokenTransfer` ? so in that sense maybe this does make sense...

Reply:
this is basically just to test that our implementations of `batchMint` `safeTransferFrom` etc functions implement this. It's just a layer to explictly make sure that because those functions implement this function, then those functions don't need to test the logic of this internal individually and instead we can just test the logic of the hook on its own once